dlp/src/test/java/dlp/snippets/InfoTypesTests.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2020 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dlp.snippets;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(JUnit4.class)
+
+public class InfoTypesTests {
+
+    private ByteArrayOutputStream bout;
+
+    private static void requireEnvVar(String varName) {
+        assertNotNull(
+                String.format("Environment variable '%s' must be set to perform these tests.", varName),
+                System.getenv(varName));
+    }
+
+    @Before
+    public void checkRequirements() {
+        requireEnvVar("GOOGLE_APPLICATION_CREDENTIALS");
+    }
+
+    @Before
+    public void setUp() {
+        bout = new ByteArrayOutputStream();
+        System.setOut(new PrintStream(bout));
+    }
+
+    @After
+    public void tearDown() {
+        System.setOut(null);
+        bout.reset();
+    }
+
+    @Test
+    public void testListInfoTypes() throws Exception {
+        InfoTypesList.listInfoTypes("supported_by=INSPECT", "en-US");
+        String output = bout.toString();
+        assertTrue(output.contains("Name") && output.contains("Display name"));
```

Comment:
protip: These tests are easier to debug if you use this format:
```java
Assert.assertThat(x, CoreMatchers.containsString("foo"));
```

The error message will be much more descriptive and easier to debug (Expected x to contain foo, but was "" instead).

Reply:
reformatted to use `Assert.assertThat()` in 66b34ca